build_extra/rust/rust.gni

```diff
@@ -62,6 +63,9 @@ template("rust_crate") {
   if (!defined(crate_name)) {
     crate_name = target_name
   }
+  if (!defined(edition)) {
+    edition = "2015"
```

Comment:
Presumably most crates will eventually be ported to 2018, so I don't think that we should default to edition 2015 -- and in general I think we do without passing --edition at all.

Don't do anything yet tho -- I'm going to apply your patch and re-run the BUILD.gn generator and see what happens.